server/services/v1/query_executor.go

```diff
@@ -0,0 +1,237 @@
+// Copyright 2022 Tigris Data, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1
+
+import (
+	"context"
+	"math/rand"
+	"sync"
+	"time"
+
+	"github.com/rs/zerolog/log"
+	api "github.com/tigrisdata/tigris/api/server/v1"
+	"github.com/tigrisdata/tigris/server/metadata"
+	"github.com/tigrisdata/tigris/server/transaction"
+	"github.com/tigrisdata/tigris/store/kv"
+	ulog "github.com/tigrisdata/tigris/util/log"
+	"google.golang.org/grpc/codes"
+)
+
+// QuerySessionExecutor is used to manage all the explicit query sessions. The execute method is executing the query.
+// The method uses the txCtx to understand whether the query is already started(explicit transaction) if not then it
+// will create a QuerySession and then will execute the query. For explicit transaction, Begin/Commit/Rollback is
+// creating/storing/removing the QuerySession.
+type QuerySessionExecutor struct {
+	sync.RWMutex
+
+	txMgr     *transaction.Manager
+	tenantMgr *metadata.TenantManager
+	versionH  *metadata.MetaVerHandler
+	tracker   *querySessionTracker
+}
+
+func NewQuerySessionExecutor(txMgr *transaction.Manager, tenantMgr *metadata.TenantManager, versionH *metadata.MetaVerHandler) *QuerySessionExecutor {
```

Comment:
Can we call it `SessionManager`?

Reply:
We can - I was thinking just to avoid the confusion that it is not only keeping transaction session but a bit more. I am fine with calling it SessionManager.